code/modules/surgery/organs/blood.dm

```diff
@@ -117,264 +73,218 @@
+					return 1
 
-	var/datum/reagent/blood/injected = get_blood(container.reagents)
+			C.blood_volume = min(C.blood_volume + round(amount, 0.1), BLOOD_VOLUME_MAXIMUM)
+			return 1
 
-	if(!istype(injected)) //Sanity..
-		return
+	AM.reagents.add_reagent(blood_id, amount, blood_data, bodytemperature)
+	return 1
 
-	if(species && species.flags & NO_BLOOD)
-		reagents.add_reagent("blood", amount, injected.data)
-		reagents.update_total()
-		return
 
-	var/datum/reagent/blood/our = get_blood(vessel)
+/mob/living/proc/get_blood_data(blood_id)
+	return
 
-	if(!istype(our)) // Do we not have blood?
-		vessel.add_reagent("blood", amount, injected.data) // This is only ran once, so we need to add back their DNA.
-		vessel.update_total()
-		..()
-		our = get_blood(vessel) // Get the new blood.
-		our.data["blood_DNA"] = dna.unique_enzymes // Give it the proper DNA we had before (src.dna)
-		our.data["blood_type"] = dna.b_type  // Proper blood type.
-		our.data["donor"] = src // Without this, it'd be random each time it was extracted or Null.
-		return
+/mob/living/carbon/human/get_blood_data(blood_id)
+	if(blood_id == "blood") //actual blood reagent
+		var/blood_data = list()
+		//set the blood data
+		blood_data["donor"] = src
+		blood_data["viruses"] = list()
 
-	if(blood_incompatible(injected.data["blood_type"],our.data["blood_type"]) )
-		reagents.add_reagent("toxin",amount * 0.5)
-		reagents.update_total()
-	else
-		vessel.add_reagent("blood", amount, injected.data)
-		vessel.update_total()
-	..()
-
-//Gets human's own blood.
-/mob/living/carbon/proc/get_blood(datum/reagents/container)
-	var/datum/reagent/blood/res = locate() in container.reagent_list //Grab some blood
-	if(res) // Make sure there's some blood at all
-		if(res.data["donor"] != src) //If it's not theirs, then we look for theirs
-			for(var/datum/reagent/blood/D in container.reagent_list)
-				if(D.data["donor"] == src)
-					return D
-	return res
-
-/mob/living/carbon/human/get_blood(datum/reagents/container)
-	if(species.exotic_blood)
-		return container.get_reagent_from_id(species.exotic_blood)
-	else
-		return ..()
+		for(var/datum/disease/D in viruses)
+			blood_data["viruses"] += D.Copy()
 
-/mob/living/carbon/proc/get_blood_name()
-	return "blood"
+		blood_data["blood_DNA"] = copytext(dna.unique_enzymes,1,0)
+		if(resistances && resistances.len)
+			blood_data["resistances"] = resistances.Copy()
+		var/list/temp_chem = list()
+		for(var/datum/reagent/R in reagents.reagent_list)
+			temp_chem[R.id] = R.volume
+		blood_data["trace_chem"] = list2params(temp_chem)
+		if(mind)
+			blood_data["mind"] = mind
+		if(ckey)
+			blood_data["ckey"] = ckey
+		if(!suiciding)
+			blood_data["cloneable"] = 1
+		blood_data["blood_type"] = copytext(src.dna.b_type,1,0)
+		blood_data["gender"] = gender
+		blood_data["real_name"] = real_name
+		if(species.blood_color)
+			blood_data["blood_color"] = species.blood_color
+		else
+			blood_data["blood_color"] = "#A10808"
```

Comment:
Don't think this is necessary; this is only applicable to humans, and this would need their species to be `null` for this to pass, which means there's a LOT bigger problems.